src/api/EpiData.ts

```diff
@@ -43,11 +44,23 @@ const ENDPOINT = process.env.EPIDATA_ENDPOINT_URL;
 
 export const fetchOptions: RequestInit = process.env.NODE_ENV === 'development' ? { cache: 'force-cache' } : {};
 
+function processResponse<T>(response: Response): Promise<T> {
+  if (response.ok) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return response.json();
+  }
+  return response.text().then((text) => {
+    throw new Error(`[${response.status}] ${text}`);
+  });
+}
+
 export function fetchImpl<T>(url: URL): Promise<T> {
   const urlGetS = url.toString();
   if (urlGetS.length < 4096) {
     // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-    return fetch(url.toString(), fetchOptions).then((d) => d.json());
+    return fetch(url.toString(), fetchOptions).then((d) => {
+      return processResponse(d);
+    });
```

Comment:
would this work (or something similar)?
```suggestion
    return fetch(url.toString(), fetchOptions).then(processResponse);
```

Reply:
explored in https://github.com/cmu-delphi/www-epivis/pull/86